lib/rucio/client/downloadclient.py

```diff
@@ -73,6 +73,134 @@ def __init__(self, _client=None, user_agent='rucio_clients', logger=None):
         self.trace_tpl['eventType'] = 'download'
         self.trace_tpl['eventVersion'] = 'api'
 
+    def download_file_from_archive(self, items, trace_custom_fields={}):
+        """
+        Download items with a given PFN. This function can only download files, no datasets.
+
+        :param items: List of dictionaries. Each dictionary describing a file to download. Keys:
+            did                 - DID string of the archive file (e.g. 'scope:file.name'). Wildcards are not allowed
+            rse                 - rse name (e.g. 'CERN-PROD_DATADISK'). RSE Expressions are not allowed
+            archive             - name of the archive from which the file should be extracted
+            base_dir            - Optional: Base directory where the downloaded files will be stored. (Default: '.')
+            no_subdir           - Optional: If true, files are written directly into base_dir and existing files are overwritten. (Default: False)
+            ignore_checksum     - Optional: If true, the checksum validation is skipped (for pfn downloads the checksum must be given explicitly). (Default: True)
+            transfer_timeout    - Optional: Timeout time for the download protocols. (Default: None)
+        :param num_threads: Suggestion of number of threads to use for the download. It will be lowered if it's too high.
+        :param trace_custom_fields: Custom key value pairs to send with the traces
+
+        :returns: a list of dictionaries with an entry for each file, containing the input options, the did, and the clientState
+                  clientState can be one of the following: ALREADY_DONE, DONE, FILE_NOT_FOUND, FAIL_VALIDATE, FAILED
+
+        :raises InputValidationError: if one of the input items is in the wrong format
+        :raises NoFilesDownloaded: if no files could be downloaded
+        :raises NotAllFilesDownloaded: if not all files could be downloaded
+        :raises RucioException: if something unexpected went wrong during the download
+        """
+        logger = self.logger
+        trace = copy.deepcopy(self.trace_tpl)
+        log_prefix = 'Extracting'
```

Comment:
log_prefix is just used once and it's missing a whitespace at the end. You could write it directly in the message or do you plan to use it more often later?

Reply:
Yes I do. Once we add transparent replica listing at some far future.

Reply:
I am also thinking about some validation of input at the first place. log_prefix is now: 'Extracting files : '